### ISSUE TITLE
CREEL_SAMPLES sampleID issues

### ISSUE THREAD

**Issue by kaijagahm** [MFEh2o/db#123]
This is a spinoff of #51: decided to deal with it separately while I finalize the rest of the stuff in that issue since there are several problems coming up with the CREEL_SAMPLES table.

Here's what I've done so far:
1. Noticed that the times in CREEL_SAMPLES are off--mismatches between the sampleID and the dateTimeSample. Here is code that demonstrates that:
```
# Set dbdir and db, and source dbUtil.R
# read in CREEL_SAMPLES as `cs`
cs <- dbTable("creel_samples")

sampleIDTimes <- str_extract(cs$sampleID, "(?<=_)\\d{4}(?=_)") %>%
  as.numeric()
dateTimeSampleTimes <- substr(cs$dateTimeSample, 12, 16) %>%
  str_replace(., ":", "") %>%
  as.numeric()
## Look at the difference:
sampleIDTimes-dateTimeSampleTimes # ok, there are actually a whole bunch of differences, so it's not going to be a straightforward fix.
```
Initially, I looked at the first few times and noticed they were all 5 hours off, but the above code demonstrates that there are a whole bunch of discrepancies. In fact, all but two rows have mismatched times.

As a way to begin to dig through this, Randi sent me the IS database file for the 2019 creel data. It's not attached because it's a csv file and github doesn't support attaching those to issues (WHY???), but the file is called "creelSamplesIS.csv" and Randi sent it to me via Slack.

Note: there are a lot of repeated rows in that CSV file because of the way Ashley entered the data. But that shouldn't matter for our purposes.

The first thing I did was check to see whether all the sampleID's in creelSamplesIS.csv appear in CREEL_SAMPLES. For some reason, none of them do.
```
is2019 <- read.csv(here("data", "inputs", "CREEL_SAMPLES_timeCheck)) %>% distinct() # remove repeat rows.
is2019$sampleID %in% cs$sampleID # none of them are!! What the heck?
```
Then I noticed that a lot of the sampleID's in `is2019` have siteID "boatLaunch" instead of "BoatLaunch" in the CREEL_SAMPLES table. I changed those and then checked again:
```
is2019 <- is2019 %>%
  mutate(sampleID = str_replace(sampleID, "boatLaunch", "BoatLaunch"))

is2019$sampleID %in% cs$sampleID # still NONE!
```

Just to get a handle on what was going on, I opened up CREEL_SAMPLES in the viewer:
```
View(cs)
```

And in the search bar, I typed in "AQ_BoatLaunch_201906" (basically just beginning to type in the first sampleID that comes up with `head(is2019$sampleID)`. 

I discovered two things. First, the problem seems to stem from what looks like dragdown error in the metadataID's in `is2019`. So maybe by correcting those I could match them up.

But second, I noticed that there seem to be near-duplicates of each sampleID. For example, in CREEL_SAMPLES, we have both `AQ_BoatLaunch_20190617_2130_fishscapesCreel.Interview.20200410` and `AQ_BoatLaunch_20190617_2130_CR_fishscapesCreel.Trailers.20200512`, the main difference being that one of these is a sampleID that pertains to the trailer sampling and one of them pertains to the interview sampling. But then I'm confused, because when I search CREEL_TRAILER_SAMPLES, I don't see any samples at all that come from lake AQ, and the samples that do appear in CREEL_TRAILER_SAMPLES have a different metadataID portion. 

The other difference between these sampleID's, aside from the metadataID portion, is that one of them contains the gear "CR", while one of them doesn't have a gear portion. It seems like we should standardize this: if the creel sampleID's are going to follow the format for fish samples, then we should add the creel gear to all of them. Or we could remove it from all of them (since it's kind of redundant, since all of them will have CR), but then we would need to make sure to document really carefully how the creel sampleID's are put together.

So basically, @Randinotte, I think this is a cry for help from me because I'm confused about how these tables are structured, and as a result I'm having trouble figuring out what's a mistake and what's just me not understanding the structure. At a minimum, we need to be more specific with our documentation, and I suspect that there are also some problems that need to be fixed.

When you have a moment, let's hop on a call and discuss this.

**Comment by kaijagahm** [comment]
Notes from our meeting today:
1. After a bunch of investigation, we've decided to trust the times and dates in the sampleID. Steps to fix this:
    a. Extract date and time from sampleID to create a new dateTimeSample column
    b. Compute the difference between the original dateTimeSample and the new dateTimeSample
    c. Subtract that difference measurement from the new dateTimeSample column to create a new dateTimeSet column
    d. Check whether the dates in dateSet and dateSample match the dates in the new dateTimeSet and new dateTimeSample columns. They very likely do, but if they don't, go ahead and overwrite them with the new values--we are trusting the sampleID.
    e. Remove the old dateTimeSet and dateTimeSample columns and rename the new ones appropriately.
    f. Make sure that the dateTimeSample and dateSample columns in all the other tables are created correctly, from the sampleID
    g. Extract all unique sample times, summarize them, and make sure they make sense.

2. I should add CR as gear in all the places it's missing. It was a mistake to have it be missing. We debated replacing the rather generic "CR" with three different values: "IN" for interview, "BO" for boat count, and "TR" for trailer count. But decided to keep it all as "CR" since the gear actually serves a function in the CREEL_FISH table, and if it were ever joined with the regular fish table, we would want a very clear way to link it back to the creel project. 
3. The near-duplicate sampleID's are normal. There are usually three sampleID's for each lake/date: one for trailer counts, one for boat counts, and one for interviews. But the boat count one has WholeShoreline as its site, so it doesn't come up as a near duplicate. These are distinguished from each other by the metadataID, which indicates what type of sample it is. The rationale between including all three of these in one CREEL_SAMPLES table is that we need somewhere to store the sample-level info, like crew, weather, etc., and we don't want to have to repeat that over and over in each of the sub-tables. Meanwhile, FLIGHTS_SAMPLES is its own separate table just because the format is so different.
